.github/workflows/ci-binder-link-comment.yaml

```diff
@@ -0,0 +1,19 @@
+name: Comment with Binder link on PRs
+
+on: pull_request_target
```

Comment:
Is there a reason we need to use `pull_request_target` instead of `pull_request`? I recall there being some possible security concerns when using `pull_request_target` but don't know if they apply in this context or not

Reply:
Yeah this is intentional. In this context `pull_request_target` is protecting us from security issues. When you use this event type the target branch is checked out rather than the PR branch. The target branch is trusted so GitHub makes the repo secrets are made available and the `GITHUB_TOKEN` has read/write. I had to do it this way in order to access tokens that can make comments.

The security concern you may be thinking of is that folks may then manually pull in the PR branch, which means untrusted code is being run with access to secrets and tokens. This isn't the case in this action. I'm not checking out the PR code or doing anything with the content of the PR, so I have no security concerns.

The trade-off here is that if we make PRs to modify this action they can't really be tried out and tested until they have been merged to `main` because the action triggered by  `pull_request_target` will read the action YAML file from `main` not the modified one in the PR.

Hopefully that addresses your concerns?